Added texture field to origins.

ThatRobin opened the pull request:
This PR creates a textures field in the origin json files which can be used to replace any GUI element with a custom texture.
To avoid missing texture issues, the default textures are also used as fallback textures if the specified identifier does not exist in the gui sprite atlas.

eggohito committed:
Aligned the extra bar icons in bar sprite sheets

eggohito committed:
Fixed origin name widget not being refreshed on screen resize

eggohito committed:
Bumped mod and Apoli version

ThatRobin committed:
Added texture field to origins.
Moved background.png to background/window.png.
Added screen.png, as a default screen background acting as the default dirt texture.

eggohito committed:
Fixed merging of origin layers

eggohito committed:
Added support for `*` in identifiers in origin layers

eggohito committed:
Tweaked how the origin screen is rendered
* The contents of the origin screen now looks a lot more seamless, where the text renders behind the window border frame and the window name plate
* Simplified how the origin name widget is rendered

eggohito committed:
Tweaked execution logic of the action in `origins:action_on_callback` power type
* The action is now only executed if the entity condition specified in the power's `condition` field is fulfilled or absent

eggohito committed:
Tweaked logic for determining whether a player has all the origins
* Now accounts for layers that do not have any options (e.g: if a layer automatically chooses an origin conditionally, and if that condition is not fulfilled)

eggohito committed:
Bumped mod and Apoli version

eggohito committed:
Fixed issue with removal of old powers from an old origin

eggohito committed:
Added compatibility with Carpet's fake players for syncing origins on join

eggohito commented on src/main/java/io/github/apace100/origins/origin/Origin.java:
```diff
@@ -33,13 +34,14 @@ public class Origin {
         .add("impact", OriginsDataTypes.IMPACT, Impact.NONE)
         .add("loading_priority", SerializableDataTypes.INT, 0)
         .add("upgrades", OriginsDataTypes.UPGRADES, null)
+        .add("textures", ApoliDataTypes.IDENTIFIER_MAP, new LinkedHashMap<>())
```
I think the name of the field should be `texture_map`, just to be consistent with the `texture_map` field of the `status_bar_texture` power type that behaves the same

ThatRobin replied:
oh yeah for sure, i forgot to check the name of the existing one, will change when i'm able to

eggohito commented on src/main/java/io/github/apace100/origins/util/TextureUtil.java:
Just a minor nitpick: I think this util. class could be generalized for checking if a texture is loaded or if a sprite exists in an atlas, similar to what I did in my [dynamic container type PR](https://github.com/apace100/apoli/pull/154/files#diff-f1c20138fd3d3a685ffeebfd1356ccbffb3b02f803ba70bdf77803a99afbf171) in Apoli

eggohito reviewed (changes requested):
Comments aside, I think this is very cool and will give users a lot of freedom in terms of customizability :D

eggohito committed:
Fixed hidden origin layers not being hidden

eggohito committed:
Fixed an untranslated message for the Avian origin

eggohito committed:
Bumped mod version

eggohito committed:
Bumped Apoli version

eggohito committed:
Bumped Apoli version (again)

eggohito committed:
Made changes to Arachnid's Master of Webs and Climbing powers
* Fixed Arachnid unable to hold its climb on sneak
* Fixed Arachnid unable to go down on sneak while climbing cobweb-like blocks
+ Added NOT sneaking condition for webbing attacked entities

eggohito committed:
Tweaked Arachnid's climbing power
* Added a set of condition that evaluates whether the entity is not sneaking and if it's horizontally colliding with a block; to keep the old behavior of climbing

eggohito committed:
Fixed issue with syncing origins

eggohito committed:
Simplified reading origins from a buffer in `SyncOriginRegistryS2CPacket`

eggohito committed:
Bumped dependencies and updated build stuff

eggohito committed:
Fixed transitive dependency issue

eggohito committed:
Fixed issue with syncing the badge registry to the client

eggohito committed:
Fixed issue with syncing badges to the client (for real this time)
For some reason, writing to a buffer directly and using the Networking v1 API works for both singleplayer and multiplayer, while using the object-oriented Networking API (v2) doesn't :thinking:

eggohito committed:
Bumped mod and Apoli versions

eggohito committed:
Bumped Apoli version (again)

eggohito committed:
Fixed issue with the result stack of recipe badge tooltips
..where the result stack is not being modified by a power that uses the `modify_crafting` power type

eggohito committed:
Fixed edge-case crash issue when removing disabled powers from origins

eggohito committed:
Bumped Loom, mod, and Apoli versions

eggohito committed:
Bumped mod and Apoli version

eggohito committed:
Fixed a crash issue caused by the `origin` entity condition type
Caused by trying to get a non-existent origin layer

eggohito committed:
Bumped mod and Apoli version

ThatRobin committed:
renamed textures to texture_map

ThatRobin committed:
removed TextureUtil in favor of Apoli's TextureUtil.

ThatRobin committed:
removed Pre-Power-Reload id logging.

ThatRobin committed:
Added texture mapping for the buttons, and implemented the Apoli TextureUtil class.

eggohito commented:
I'll be closing this PR for now as I'll be taking over (and also to fix the funky commits.) I'll make another PR at some point
